src/index.ts

```diff
@@ -163,6 +167,13 @@ export function createProxyResolver(params: ProxyAgentParams) {
 			return;
 		}
 
+		if (typeof hostname === 'string' && configNoProxy(hostname, String(parsedUrl.port || defaultPort))) {
```

Comment:
To align with the following from the PR for VSCode:
```ts
			const newNoProxy = (this.configurationService.getValue<string[]>('http.noProxy') || []).map((item) => item.trim()).join(",")
 				|| (process.env['no_proxy'] || process.env['NO_PROXY'] || '').trim() || undefined;
```

It would make sense to check if there are any config entries and in that case ignore the env variable. (This also aligns well with how a proxy from the user settings overrides any other values.)